packages/repository/src/repositories/relation.factory.ts

```diff
@@ -47,9 +47,9 @@ export function hasManyRepositoryFactory<SourceID, T extends Entity, ID>(
     case RelationType.hasMany:
       const fkConstraint = {[relationMetadata.keyTo]: sourceModelId};
 
-      return new DefaultHasManyEntityCrudRepository(
-        targetRepository,
-        fkConstraint,
-      );
+      return new DefaultHasManyEntityCrudRepository<
```

Comment:
Generics are set manually since typescript would crap out if the instance being returned was first assigned to a variable